server/src/main/java/org/candlepin/guice/CandlepinContextListener.java

```diff
@@ -291,6 +294,35 @@ private void destroySubsystems() throws Exception {
         this.loggerListener.contextDestroyed();
     }
 
+    private void waitForArtemisJobThreadsToFinish() {
```

Comment:
If this is specifically for the job system, we should move this to somewhere that makes a bit more sense. Either in jobManager.shutdown or, and probably best in my opinion, ArtemisContextListener.shutdown (which gets hit automatically via cpmContextListener.shutdown above)

Reply:
Same as above answer, by the time we get to `ArtemisContextListener.shutdown`, the sessions are already closed. We want the sessions to keep open for the message commits to happen first